agent/rpc_to.go

```diff
@@ -71,12 +72,48 @@ func (a *orbAgent) sendGroupMembershipReq() error {
 	if token := a.client.Publish(a.rpcToCoreTopic, 1, false, body); token.Wait() && token.Error() != nil {
 		return token.Error()
 	}
-
 	return nil
 }
 
+func (a *orbAgent) sendGroupMembershipReq() error {
+	defer a.retryGroupMembershipRequest()
+	return a.sendGroupMembershipRequest()
+}
+
+func (a *orbAgent) retryGroupMembershipRequest() {
+	if a.groupRequestTicker == nil {
+		a.groupRequestTicker = time.NewTicker(retryRequestFixedTime * retryRequestDuration)
+	}
+	go func() {
+		defer a.groupRequestTicker.Stop()
+		for calls := 1; calls <= retryMaxAttempts; calls++ {
+			select {
+			case <-a.groupRequestSucceeded:
+				a.groupRequestTicker.Stop()
+				return
+			case _ = <-a.groupRequestTicker.C:
+				a.logger.Info("agent has not received any group membership from fleet, re-requesting")
+				duration := retryRequestFixedTime + (calls * retryDurationIncrPerAttempts)
+				a.groupRequestTicker.Reset(time.Duration(duration) * retryRequestDuration)
+				err := a.sendGroupMembershipRequest()
+				if err != nil {
+					a.logger.Error("failed to send group membership request", zap.Error(err))
+					return
+				}
+			}
+		}
+		a.logger.Warn(fmt.Sprintf("retried %d times and still got no response from fleet", retryMaxAttempts))
```

Comment:
let's add in this log message that this is specific to group request (as opposed to policy)

Reply:
actually, I am looking at the result of the log, and we have the method caller there, don't you think it would be redundant?

Reply:
hm i'd need to see an example, do you mean the call stack? i think I would rather have it directly in the message though, i think the method may not show up in some contexts